src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -137,9 +145,37 @@ public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connec
         }
 
         /// <summary>
-        /// Open a connection with the specified connection details
+        /// Validates the given ConnectParams object. 
+        /// </summary>
+        /// <param name="connectionParams">The params to validate</param>
+        /// <returns>A ConnectionCompleteParams object upon validation error, 
+        /// null upon validation success</returns>
+        public ConnectionCompleteParams ValidateConnectParams(ConnectParams connectionParams)
+        {
+            string paramValidationErrorMessage;
+            if (connectionParams == null)
+            {
+                return new ConnectionCompleteParams
+                {
+                    Messages = SR.ConnectionServiceConnectErrorNullParams
+                };
+            }
+            if (!connectionParams.IsValid(out paramValidationErrorMessage))
+            {
+                return new ConnectionCompleteParams
+                {
+                    OwnerUri = connectionParams.OwnerUri,
+                    Messages = paramValidationErrorMessage
+                };
+            }
+
+            // return null upon success
+            return null;
+        }
+
+        /// <summary>
+        /// Open a connection with the specified ConnectParams
         /// </summary>
-        /// <param name="connectionParams"></param>
         public async Task<ConnectionCompleteParams> Connect(ConnectParams connectionParams)
```

Comment:
An overall comment on the Connect & Disconnect methods: they were already large and have now hit the "too large" level. Most of the functionality can be split into discrete methods and the main method flow will then be a sequence of steps that you call, instead of the complex logic right now. Please work on this and see if it reveals any common code you can combine :)